index.js

```diff
@@ -138,8 +138,8 @@ HtmlWebpackPlugin.prototype.apply = function (compiler) {
       .then(function (html) {
         var pluginArgs = {html: html, assets: assets, plugin: self, outputName: self.childCompilationOutputName};
         return applyPluginsAsyncWaterfall('html-webpack-plugin-before-html-processing', pluginArgs)
-          .then(function () {
-            return pluginArgs.html;
+          .then(function (newPluginArgs) {
+            return newPluginArgs && newPluginArgs.html || pluginArgs.html;
```

Comment:
No if you would explicitly return an empty string it would fallback to the pluginArgs (strange use case but still a unexpected behaviour)